src/charm.py

```diff
@@ -1,69 +1,159 @@
+        self._port = MONGODB_PORT
+
         self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.start, self._on_start)
 
-    def _on_install(self, _):
-        """Install prerequisites for the application"""
+    def _on_install(self, _) -> None:
+        """Handle the install event (fired on startup).
+
+        Handles the startup install event -- installs updates the apt cache, installs MongoDB.
+        """
         self.unit.status = MaintenanceStatus("installing MongoDB")
         self._add_mongodb_org_repository()
         self._install_apt_packages(["mongodb-org"])
-        self.unit.status = WaitingStatus("Waiting to start MongoDB")
 
-    def _add_mongodb_org_repository(self):
+    def _on_config_changed(self, _: ops.charm.ConfigChangedEvent) -> None:
+        """Event handler for configuration changed events."""
+        # TODO
+        # - update existing mongo configurations based on user preferences
+        # - add additional configurations as according to spec doc
+        with open("/etc/mongod.conf") as mongo_config_file:
+            mongo_config = yaml.safe_load(mongo_config_file)
+
+        machine_ip = str(self.model.get_binding(PEER).network.bind_address)
+        bind_ips = "localhost," + machine_ip
+        mongo_config["net"]["bindIp"] = bind_ips
+        if "replication" not in mongo_config:
+            mongo_config["replication"] = {}
+
+        mongo_config["replication"]["replSetName"] = "rs0"
+        with open("/etc/mongod.conf", "w") as mongo_config_file:
+            yaml.dump(mongo_config, mongo_config_file)
+
+        self.unit.status = ops.model.ActiveStatus()
+
+    def _on_start(self, event: ops.charm.StartEvent) -> None:
+        """Enables MongoDB service and initializes replica set.
+
+        Args:
+            event (ops.charm.StartEvent): The triggering start event.
+        """
+        if not self.unit.is_leader():
+            return
+
+        # start mongo service
+        self.unit.status = MaintenanceStatus("enabling MongoDB")
+        if not service_running("mongod.service"):
+            logger.debug("enabling MongoDB")
+            mongod_enabled = service_resume("mongod.service")
+            if not mongod_enabled:
+                logger.error("failed to enable mongod.service")
+                self.unit.status = BlockedStatus("failed to enable mongod.service")
+                return
+
+        try:
+            self._open_port_tcp(self._port)
+        except Exception:
+            self.unit.status = BlockedStatus("failed to open TCP port for MongoDB")
+            return
+
+        if not self.mongo.is_ready():
+            self.unit.status = WaitingStatus("Waiting for MongoDB Service")
+            event.defer()
+            return
+
+        # initialize replica set
+        self.unit.status = MaintenanceStatus("initializing MongoDB replicaset")
```

Comment:
```suggestion
        self.unit.status = MaintenanceStatus("initialising MongoDB")
```

Canonical uses British English for spelling ;-)